common.eai

```diff
@@ -9753,37 +9756,43 @@ function AddHarassUnittype takes integer groupnum, integer harassnum, group g, g
   local unit u = null
   local real player_sum = 0
   set cg = SelectNumberOfId(cg, qty, old_id[uid])
+  set cg = SelectByUnitFree(cg, true)
+  set cg = SelectByUnitStandard(cg, true)  //Reduce usage IsUnitInGroup
   loop
     set u = FirstOfGroup(cg)
     exitwhen u == null
-    call GroupAddUnit(unit_harassing, u)
-    call RemoveGuardPosition(u)
     call GroupAddUnit(harasser,u)
     set player_sum = player_sum + GetUnitStrength(u)
     call GroupRemoveUnit(cg,u)
     call GroupRemoveUnit(g,u)
   endloop
   call DestroyGroup(cg)
   set cg = null
-  call SaveReal(additional_info, key, START_STRENGTH, player_sum)
+  call SaveReal(additional_info, key, START_STRENGTH_TEMP, player_sum)
   return harasser
 endfunction
 
 function StartHarass takes integer groupnum, integer harass_target, boolean avoid_towers, real strength_limit, real flee_percent, integer flee_number, real lx, real ly returns nothing
   local integer i = 0
   local integer t = 0
-  local group harasser = CreateGroup()
-  local group g = CreateGroup()
+  local group harasser = null
+  local group g = null
   local integer key = groupnum
   local real strength = 0
+  local real getstrength = 0
   local integer qty = 0
   local integer uid = 0
+  if HaveSavedHandle(additional_info, key, HARASS_GROUP) then
+    call Trace("StartHarass : the groupnum harass runing , not start new harass job" )
+    return
+  endif
+  call Trace("StartHarass : get harass unit" )
+  set harasser = CreateGroup()
+  set g = CreateGroup()
+  set getstrength = LoadReal(additional_info, key, START_STRENGTH)
```

Comment:
Similarly to above this will eventually result in infinite strength and the group never running away.  Just leave at 0.